cdc/api/middleware/middleware.go

```diff
@@ -185,3 +187,56 @@ func CheckServerReadyMiddleware(capture capture.Capture) gin.HandlerFunc {
 		}
 	}
 }
+
+// AuthenticateMiddleware authenticates the request by query upstream TiDB.
+func AuthenticateMiddleware(capture capture.Capture) gin.HandlerFunc {
+	return func(ctx *gin.Context) {
+		m, err := capture.GetUpstreamManager()
+		if err != nil {
+			_ = ctx.Error(err)
+			ctx.Abort()
+			return
+		}
+		up, err := m.GetDefaultUpstream()
```

Comment:
it's better to support multi-upstream here

Reply:
We will support this feature when implementing multiple upstream in the future.